activemodel/test/cases/type/decimal_test.rb

```diff
@@ -66,6 +66,8 @@ def test_changed?
         assert_not type.changed?(5.0, 5.0, "5.0")
         assert_not type.changed?(-5.0, -5.0, "-5.0")
         assert_not type.changed?(5.0, 5.0, "0.5e+1")
+        assert_not type.changed?(BigDecimal::NAN, BigDecimal::NAN, BigDecimal::NAN)
+        assert type.changed?(BigDecimal::NAN, ::Float::NAN, ::Float::NAN)
```

Comment:
This only pass because you use the `NAN` constants which are singletons.

if you use `0.0/0` to generate a new `NaN` instance, the test will fail.

Reply:
@byroot Ah yeah I forget that 🤦‍♂️ I'll take a look on this tonight

Reply:
@byroot I wrote tests for both scenarios, not sure if it's really required since a test with `0.0 / 0.0` would be enough.

Reply:
Yeah, just `0.0/0.0` and `BigDecimal('0.0') / 0` is enough.